dev-docs/RFCs/v8.1/gpu-accelerated-polygon-clipping.md

```diff
@@ -0,0 +1,153 @@
+# RFC: GPU Accelerated Polygon Clipping
+
+* Authors: Ravi Akkenapally
+* Date: Feb 03, 2020
+* Status: **Draft**
+
+## Overview
+
+Finding weather a point resides inside a polygon or not (in 2D space) has complexity Of `(N * M)`,  where `N` is the number of points and `M` is number of polygon edges. When performing this on large number of points takes a long time. This RFC proposes a GPU accelerated approach that can be implemented within `WebGL` limitation.
+
+### Existing Research Work
+
+Finding a point inside a polygon or not is also referred as PIP test. There has been lot of research work recently, to accelerate the PIP test using GPU.
+
+
+Following is the list of papers I found and explored :
+
+1. Speeding up Large-Scale Point-in-Polygon Test Based Spatial Join on GPUs ([Paper](http://www-cs.ccny.cuny.edu/~jzhang/papers/bigspaital_cr.pdf))
+
+2. A cell-based point-in-polygon algorithm suitable for large sets of points ([Paper](https://www.sciencedirect.com/science/article/pii/S0098300401000371?via%3Dihub))
+
+3. GPU Rasterization for Real-Time Spatial Aggregation over Arbitrary Polygons ([Paper](http://www.vldb.org/pvldb/vol11/p352-zacharatou.pdf))
+
+4. A Simple and fast hardware-accelerated point-in-polygon test ([Paper](https://pdfs.semanticscholar.org/04a4/7b99d57cdf81bbd58a8dc21c2d15ac528855.pdf))
+
+All of above work uses more advanced Graphics APIs (such as OpenGL Compute Shaders, CUDA, SSBO, etc) and or not supported either by WebGL1 or WebGL2. This RFC proposes a method that can be implemented with in WebGL1 and WebGL2 limitations.
+
+
+## Proposed Approach
+
+A texture based filtering can be implemented to perform PIP test on GPU. The idea is, user provided polygon(s) is triangulated and rendered to an offline frame buffer. And then, given points are run through transform feedback loop, each point is translated into the texture space and above texture is sampled, result of this sample determines if the point is inside or outside the polygon.
+
+
+## Proposed New Components
+
+### `texturefilter` (ShaderModule)
```

Comment:
I'm not sure it's useful to factor this out right off the bat. I'd just keep it inside the `GPUPolygonFilter` until we find another usecase for it.

Reply:
The [MaskEffect](https://github.com/uber/deck.gl/pull/4161) in deck.gl is going to use this.